### PR TITLE
MDEV-32004: Parse error in mtr tests when using rpl_check_server_ids

### PR DESCRIPTION
**Pull request by an3l**
* Make `rpl_check_server_ids` argument work again.
* Remove extra connections namely `server_[1,2,..]_1`, that were created during initalization in `rpl_init.inc`, recreated in `rpl_reconnect.inc` and removed in `rpl_end.inc`.
They are dead connections, not used anywhere in `rpl` suite.
* Cosmetic fixes
  * Fix errors in `rpl_init.inc` regarding the comments
    * sed 's/.cfg/.cnf/'
    * Verifying server_id is not by default but with using `rpl_check_server_ids` parameter.
    * Add comment why extra connections `server_[i,..,N]_1` are needed, since there is no usage of env variables `[MASTER,SLAVE]_PORT1` except in `rpl_show_slave_hosts.test`.
* Other cleanup
  * More debug messages,
  * Missing arguments
  * Spaces.

* `rpl_init.inc` flow chart
image

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch.*
- [x] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

**Review by knielsen — APPROVED** [review]
Thanks, Anel, looks good.

 - Kristian.

**Review by knielsen — APPROVED** [review]
Looks good, thanks.

File: mysql-test/include/rpl_change_topology.inc [hunk 1]
```diff
@@ -83,12 +83,12 @@
 #   server N is a slave, then the N'th number is the master of server
 #   N. If server N is not a slave, then the N'th number is just spaces
 #   (so in fact it is not a number). For example, if $rpl_topology is
-#   '1->2,2->3,3->1,2->4,5->6', then $rpl_master_list is '3122 6'.
+#   '1->2,2->3,3->1,2->4,5->6', then $rpl_master_list is '3122 5'.
```
Comment (knielsen) [inline mysql-test/include/rpl_change_topology.inc:86]: Nice catch! :-)

**Review by knielsen — APPROVED** [review]
Looks fine.

 \- Kristian.

File: mysql-test/suite/rpl/r/rpl_init_test_assert.result [hunk 2]
```diff
@@ -0,0 +1,258 @@
+include/master-slave.inc
+==== BEGIN include/master-slave.inc ====
+ con='default' warn='1' qlog='1' rlog='1' aborterr='1'
+.==== BEGIN include/rpl_init.inc [topology=1->2] ====
+. con='default' warn='1' qlog='1' rlog='1' aborterr='1'
+---- Check input ----
+MASTER_MYPORT='16000' SLAVE_MYPORT='16001' MASTER_MYPORT1='' SLAVE_MYPORT1=''
```
[the remark this answers is not in the document]
Comment (bnestere) [reply mysql-test/suite/rpl/r/rpl_init_test_assert.result:7]: You'd have to add the `replace_result` before _each_ of the actual commands that outputs a host/port. _But_ then we'd lose the real host/port when actually running the test with `rpl_debug=1` (and we would want to keep it). So you'd have to add another option to mask the host/port numbers of debug output. I think it's a good idea to have the test though :)

File: mysql-test/suite/rpl/t/rpl_init_test_assert.test [hunk 3]
```diff
@@ -0,0 +1,8 @@
+--source include/have_binlog_format_row.inc
```
Comment (bnestere) [inline mysql-test/suite/rpl/t/rpl_init_test_assert.test:1]: I don't think we have a convention yet for tests that validate mtr `.inc` files.. but I'm thinking we should, to differentiate that the test shouldn't be confused with a test of a server feature (e.g the variable `init_slave`, in this case). Perhaps changing the prefix from `rpl_` to `mtr_`? @andrelkin @knielsen any thoughts?

**Review by bnestere — COMMENTED** [review]
Hey Anel! Thanks for the astute cleanup :) See my few notes.

File: mysql-test/suite/rpl/r/rpl_init_test_assert.result (hunk 2 above)
[the remark this answers is not in the document]
Comment (an3l) [reply mysql-test/suite/rpl/r/rpl_init_test_assert.result:7]: It is not the scope of this MDEV. I will remove the file from commit and attach in JIRA this debug output just for the reference.

File: mysql-test/include/rpl_for_each_slave.inc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # ==== Purpose ====
 #
-# Execute a .inc file once for each server that was configured as a
-# slave by rpl_init.inc
+# Execute a start_slave.inc or stop_slave.inc file once for each server,
+# that was configured as a slave by rpl_init.inc
```
[the remark this answers is not in the document]
Comment (an3l) [reply mysql-test/include/rpl_for_each_slave.inc:4]: There are only 2 files that are using this include, I would suggest to say about those as it is.

File: mysql-test/include/wait_for_slave_param.inc [hunk 5]
```diff
@@ -85,36 +88,44 @@ if ($_slave_check_configured == 'No such row')
...
 --let $_slave_timeout_counter= $_slave_timeout$_wait_for_slave_param_zero
 --let $_slave_continue= 1
-while ($_slave_continue)
-{
-  --let $_show_slave_status_value= query_get_value("SHOW SLAVE STATUS", $slave_param, 1)
 
-  # Check if an error condition is reached.
-  if (!$slave_error_param)
+# If slave is not already stopped
+if ($slave_error_param != "slave stopped")
```
[the remark this answers is not in the document]
Comment (an3l) [reply mysql-test/include/wait_for_slave_param.inc:93]: I will remove this part in new MDEV

File: mysql-test/include/rpl_for_each_slave.inc (hunk 4 above)
[the remark this answers is not in the document]
Comment (bnestere) [reply mysql-test/include/rpl_for_each_slave.inc:4]: That's true, but it's good to advertise how it works, along with how it's used. If I had come across this file earlier :sweat_smile:, I'd have used it for a few things to run similar tasks across all slaves. And if it was described as execute only a `start_slave.inc` or `stop_slave.inc`, I wouldn't know to look further and use it for the general purpose.

**Review by bnestere — APPROVED** [review]
Looks good, thanks @an3l!

**Pull request merged**